aiida/orm/implementation/general/node.py

```diff
@@ -1128,9 +1139,12 @@ def iterextras(self):
             # added (in particular, we do not even have an ID to use!)
             # Return without value, meaning that this is an empty generator
             return
-            yield  # Needed after return to convert it to a generator
-        for _ in self._db_iterextras():
-            yield _
+            yield # Needed after return to convert it to a generator
+        for extra in self._db_iterextras():
+            # Don't return if key == hash
+            if extra[0] == 'hash':
```

Comment:
Totally agree. I had to change some of the tests which check setting extras because they didn't expect the automatically-generated ``_aiida_hash``, but other than that "un-hiding" the extra worked fine.

Where in the documentation should we put that ``.. note::`` box?